lib/instrumentation/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+module.exports = function initialize(_agent, elastic, _moduleName, shim) {
+  shim.setDatastore(shim.ELASTICSEARCH)
+  shim.setParser(queryParser)
+
+  shim.recordQuery(elastic.Transport.prototype, 'request', function wrapQuery(shim, _, __, args) {
+    const ctx = this
+    return {
+      query: JSON.stringify(args?.[0]),
+      promise: true,
+      opaque: true,
+      inContext: function inContext() {
+        getConnection.call(ctx, shim)
+      }
+    }
+  })
+}
+
+function queryParser(params) {
+  params = JSON.parse(params)
+  const path = params.path.split('/')
```

Comment:
ok after a few interations and comparing to our other language agents I think this is what we should do:

 1. Split the path and find the first path segment that starts with `_`, that is the operation. If it only contains 1 path segment the operation is index.
 2. I think _doc should also be translated to index but with an extra segment as the operation.  Since you are creating a new index on an existing with an id. so if you do `index1/_doc/id1` The operation is `index1/index/id1/create`.
 3. In the case of _doc or index we should map the http method to an action: head(exists), put(create), delete(delete).
 4. collection is the index when it exists in path. If it does not then it should be all.

Reply:
It looks like the Go agent does this kind of fine-tuning on _doc and _search: https://github.com/newrelic/go-agent/blob/master/v3/integrations/nrelasticsearch-v7/nrelastic.go#L94

Reply:
4. according to the agent spec, when the collection name isn't known, it should be recorded as `other`.

Reply:
In this case, I've set it to `any`, which more accurately reflects the "find this in any index" scenario. I'll prep a PR in specs.